### PR TITLE
fix(events): fix the qr code scan in event registration

### PR DESCRIPTION
**Pull request by olros**
Lint, test and build
## Description

Comments/issues/screenshots:
- Converted to Typescript #243 
- Swapped out the package used for qr scanning
- Fixed bug where checking a participant wouldn't be visible in the UI

image

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] The PR includes a `closes #issueID`
- [x] The PR includes a picture showing visual changes

## Typescript convert checklist

- [x] All relative imports has been converted to absolute imports
- [x] All HOC's have been replaced with Hooks
- [x] Uses theme.spacing() instead of custom padding
- [x] Uses theme.breakpoints instead of custom media queries

### CODE REVIEW

File: src/containers/EventRegistration/index.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    color: theme.palette.colors.text.light,
+  },
+  title: {
+    color: theme.palette.colors.text.main,
+  },
+}));
+
+export type ParticipantCardProps = {
+  user: UserRegistration;
+  markAttended: (n: string, a: boolean) => void;
```
Comment (Zenjjim) [inline src/containers/EventRegistration/index.tsx:110]: hva står n og a for. Bruk mer beskrivende navn

File: src/containers/EventRegistration/index.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        <Typography className={classes.lightText}>Ingen påmeldte.</Typography>
+      )}
+    </>
+  );
+
+  return (
+    <Navigation banner={<div className={classes.top}></div>} fancyNavbar>
+      <Helmet>
+        <title>{eventName} - registering - TIHLDE</title>
+      </Helmet>
```
Comment (Zenjjim) [inline src/containers/EventRegistration/index.tsx:239]: Haha uff

File: src/containers/EventRegistration/index.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+          <div>
+            <QrReader onError={handleError} onScan={handleScan} showViewFinder={true} style={{ width: '100%' }} />
+            <br />
+            <Typography variant='body1'>QR-scanning på iOS støttes kun i Safari</Typography>
+          </div>
+        )}
+      </Paper>
+      <Snackbar onClose={handleSnackbarClose} open={snackbarOpen} TransitionComponent={Slide}>
+        <SnackbarContent
+          className={error ? classNames(classes.snackbar, classes.snackbar_error) : classNames(classes.snackbar, classes.snackbar_success)}
```
Comment (Zenjjim) [inline src/containers/EventRegistration/index.tsx:265]: `className={classNames(classes.snackbar, error ? classes.snackbar_error : classes.snackbar_success)}`

File: yarn.lock [hunk 4]
```diff
@@ -1852,6 +1852,13 @@
   dependencies:
```
Comment (Zenjjim) [inline yarn.lock:1852]: du kan godt legge inn yarn.lock i .gitignore :D

**Review by Zenjjim — CHANGES_REQUESTED** [review]

**Comment by Zenjjim** [comment]
1 runde QA ferdig

File: yarn.lock (hunk 4 above)
Comment (olros) [reply yarn.lock:1852]: https://classic.yarnpkg.com/en/docs/yarn-lock/#toc-check-into-source-control

**Review by Zenjjim — APPROVED** [review]

**Pull request merged**